subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio.c

```diff
@@ -1019,6 +1018,14 @@ void *radio_ccm_rx_pkt_set(struct ccm *ccm, u8_t phy, void *pkt)
 #endif /* CONFIG_HAS_HW_NRF_RADIO_BLE_CODED */
 #endif /* CONFIG_BT_CTLR_PHY_CODED */
 	}
+
+#if !defined(CONFIG_SOC_NRF52832) && \
+	(CONFIG_BT_CTLR_DATA_LENGTH_MAX < ((RADIO_PDU_LEN_MAX) - 4))
+	u8_t max_len = (NRF_RADIO->PCNF1 & RADIO_PCNF1_MAXLEN_Msk) >>
```

Comment:
Isn't it better to accept this as an argument instead?
You are now dependent on the order of these functions:

```
		radio_pkt_configure(8, (max_rx_octets + 4), (phy << 1) | 0x01);

		radio_pkt_rx_set(radio_ccm_rx_pkt_set(&lll->ccm_rx, phy,
						      node_rx->pdu));
```

Reply:
As the packet configure interface is the right place to pass the max length, and since the value is available in a global location, it would be wasted CPU cycles to pass as parameter. Other functions like tIFS switch interface do depend on order of hal function calls, hence this will not be an exception.

Reply:
Reading a HW register instead is definitely wasted CPU cycles. HW peripheral reads are slower than normal memory read, and the place where you call this function you have the right argument available. Leave it to the compiler to optimize the register values.

Reply:
I tend to agree with @joerchan here from a cursory look, but consistency before anything else.

Reply:
Compiler does not optimize across c files. Remember that the added code is conditionally compiled in and compiler compiling the caller does not know that. Caller will always pass parameter if I pass as parameter. For 251 bytes PDUs the assignment of MAXPACKETSIZE is not necessary as CCM reset value is 251 bytes. Without parameter passing, the  builds with 251 bytes configured are optimal.

Reply:
This said, yes, LDR instruction use more cycles but looking at the compiler generated code they are interleaved with other LDR, STR or register operations, and my belief is, instruction pipeline would balance out any CPU cycle overhead.

Conditionally compiling a function signature is another option (used elsewhere in the controller), it was a personal choice here to avoid it, for a simple code block as this. 

There is a lot of scope to optimize the controller code, both for space and speed, but the effort required is high.

Do you know why LTO is not used in Zephyr? (IIRC, there used to be Kconfig before?)

Reply:
For the record, I think I agree with @joerchan. But agree also that finding the most optimal code here requires a lot more work, which should/could be done at a later time.
On the subject of optimization (but not really related to this PR): the parameter 'phy' is not used for the nRF51-series.